### PR TITLE
Surface latest sync status for connections.

### PR DESCRIPTION
**Pull request by davinchia**
## What
This solves #2332.

TLDR: Right now there is no summary of a connector's health - the status of a connection's last sync - within the webapp. This change surfaces this information so the UI can take advantage of it.

## How
- Add the `latestSyncJobStatus` field to the `WbConnectionRead` struct. Take the chance to rename the `lastSync` field to `latestSyncJobCreatedAt` to better reflect what the field captures. `lastSync` was confusing as I expect it to return the Id/object of the last sync and not the last sync's timestamp. 
- We keep the current expectation when setting status - the same job that sets the timestamp field is used to set the status field. 
- We surface the job's status as is. This gives the UI flexibility on displaying the various statuses.

Since the 'latest' sync job is determined by the `JobHistoryHandler`, I buffed up the job history handler test cases to make sure we do always return the latest sync job.

Some minor refactors as I made changes.

## Pre-merge Checklist
- [x] UI change

## Recommended reading order
1. config.yaml
2. WebBackendConnectionsHandler.java
3. various test files

### CODE REVIEW

**Review by cgardens — APPROVED** [review]
This looks good! I would add to the checklist in the description that a UI change needs to happen before we can merge this. Can you create a quick frontend issue with what needs to be done? Basically you just need to link the original issue and describe the mapping of statuses to the colors you mentioned to me yesterday.
